### PR TITLE
Fixed the second controller doesn't show up problem

### PR DESCRIPTION
**Pull request by daoshengmu**
Per #977. We find the second controller does not show up intermittently, and I notice it is because the `model` vector of `ControllerContainer.cpp` is not resized properly.

### CODE REVIEW

File: app/src/main/cpp/ControllerContainer.cpp [hunk 1]
```diff
@@ -44,7 +44,7 @@ struct ControllerContainer::State {
   }
 
   void SetUpModelsGroup(const int32_t aModelIndex) {
-    if (models.size() >= aModelIndex) {
+    if (aModelIndex >= models.size()) {
```
Comment (bluemarvin) [inline app/src/main/cpp/ControllerContainer.cpp]: Good catch.

File: app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp [hunk 2]
```diff
@@ -709,10 +715,16 @@ struct DeviceDelegateOculusVR::State {
...
+  bool IsOculusQuest() const {
     return deviceType >= VRAPI_DEVICE_TYPE_OCULUSQUEST_START && deviceType <= VRAPI_DEVICE_TYPE_OCULUSQUEST_END;
   }
 
+  bool Is6DOF() const {
+    // ovrInputHeadsetCapabilities is unavailable in Oculus mobile SDK,
+    // the current workaround is checking if it is Quest.
+    return IsOculusQuest();
+  }
+
```
Comment (bluemarvin) [inline app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp]: I find having two different `Is6DOF()` functions confusing. Also is it possible to use a Go 3DoF controller with the Quest?

Comment (daoshengmu) [reply app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp]: One of `Is6DOF()` is for controllers, and the other one is for headsets. Currently, we only use `ControllerState.Is6DOF()` for getting the capability from controllers. We didn't have `Is6DOF()` requirement for headsets yet, I just prepare it for the future. Ideally, `ovrInputHeadsetCapabilities` should tell us the capability  flag headsets, but it doesn't work right now, and I have checked it is the same with OVR mobile samples.

**Comment by daoshengmu** [comment]
> I think this code would benefit from some clean up. I wonder if we could make it more flexible by keying off the reported caps instead of the type? And only use the type for the controller model file mapping?

yep. That's what I was doing. Probably, I am not doing good enough :)

**Pull request merged**